### PR TITLE
OIDC

### PR DESCRIPTION
**Pull request by johakoch**

### CODE REVIEW

**Comment by johakoch** [comment]
New **Beta** feature: `beta_oidc` block for configuring OIDC (authorization code flow).

File: config/oidc.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+func (o OIDC) GetRedirectURI() *string {
+	return o.RedirectURI
+}
+
+func (o OIDC) GetTokenEndpointAuthMethod() *string {
+	return o.TokenEndpointAuthMethod
+}
+
+// OpenidConfiguration represents an OpenID configuration (.../.well-known/openid-configuration)
+type OpenidConfiguration struct {
```
Comment (malud) [inline config/oidc.go:103]: This object, the related obj `OidcConfig` and logic to retrieve the configuration data should be moved to the `oauth2` package. Its to just to much non hcl/configload related context.

File: oauth2/oauth2.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+	return tokenResponse, tokenResponseData, accessToken, nil
+}
+
+// OAuth2AcClient represents an OAuth2 client using the (plain) authorization code flow.
+type OAuth2AcClient struct {
+	*AbstractAcClient
+}
+
+// NewOAuth2AC() creates a new OAuth2 Authorization Code client.
```
Comment (malud) [inline oauth2/oauth2.go:217]:
```suggestion
// NewOAuth2AC creates a new OAuth2 Authorization Code client.
```

**Review by malud — CHANGES_REQUESTED** [review]

File: eval/lib/oauth2.go [hunk 3]
```diff
@@ -39,34 +34,44 @@ func NewOAuthAuthorizationUrlFunction(oauth2Configs []*config.OAuth2AC, verifier
...
 			}
 
 			query := oauthAuthorizationUrl.Query()
 			query.Set("response_type", "code")
-			query.Set("client_id", oauth2.ClientID)
-			query.Set("redirect_uri", *oauth2.RedirectURI)
-			if oauth2.Scope != nil {
-				query.Set("scope", *oauth2.Scope)
+			query.Set("client_id", oauth2.GetClientID())
+			query.Set("redirect_uri", *oauth2.GetRedirectURI())
```
Comment (malud) [inline eval/lib/oauth2.go:50]: GetRedirectURI() should return a dereferenced string.

**Review by malud — CHANGES_REQUESTED** [review]

**Review by malud — APPROVED** [review]

**Pull request merged**